scripts/rhv_tier_analyzer.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+
+import argparse
+import subprocess
+import sys
+
+PROVIDER = 'rhv_cfme_integration'
+TEST_PATH = '../cfme/tests'
+TIER_MARKER = '-m rhv1 or rhv2 or rhv3'
+
+
+def check_virtualenv():
+    """Check if we are in virtualenv and if not, exit."""
+    if not hasattr(sys, 'real_prefix'):
+        print('You must activate CFME virtualenv in oder to run this script.')
+        sys.exit(1)
+
+
+def get_pytest_output(use_tier_marker=False):
+    """Get string output of pytest command."""
+    print('Getting pytest output {} tier marker.'.format('with' if use_tier_marker else 'without'))
+    args = ['pytest', TEST_PATH, '--collect-only', '--long-running', '--use-provider', PROVIDER]
```

Comment:
One way to further limit the collection here is to concatenate '-m uses_testgen' into the markers list, so that only test cases parameterized on providers are collected. This may also reduce how many functions are parsed out in the function below, or eliminate the need altogether to first filter by PROVIDER in the line.